src/Features/CSharp/Portable/CodeFixes/GenerateType/GenerateTypeCodeFixProvider.cs

```diff
@@ -58,7 +59,32 @@ protected override bool IsCandidate(SyntaxNode node)
 
         protected override SyntaxNode GetTargetNode(SyntaxNode node)
         {
-            return ((ExpressionSyntax)node).GetRightmostName();
+            var qualified = node.Parent as QualifiedNameSyntax;
+            if (qualified != null)
+            {
+                return GetRightMostName(qualified);
+            }
+
+            var memberAccess = node.Parent as MemberAccessExpressionSyntax;
+            if (memberAccess != null)
+            {
+                return GetRightMostName(memberAccess);
+            }
+
+            return (node as ExpressionSyntax)?.GetRightmostName();
```

Comment:
Ditto as the comment I left on the VB side. Why is ```GetRightMostName``` needed below? What makes it different from the ```GetRightmostName``` extension method? Why can't it replace the extension method? Why didn't use fix the casing of the extension method name in this PR? :smile: